File: FilterDesign/WindowEffect/test_original_filter_in_c.m
fps = 30;

LPF = 0.5;
HPF = 3.33;

n = -511:512;
n0 = 0;
imp = (n==n0);


%% Original filter in C
tmp_N = 512;
tmp_n = 0:1:tmp_N-1;
freq = ( n .* fps) ./ tmp_N;
imp_rep = bandpassfilter(imp, 128, LPF, HPF,fps);
imp_rep =[zeros(1,2048) imp_rep' zeros(1,2048)];
filter_analysis_imp(imp_rep,fps);
title("Ideal BPF with windowing");

File: Ch7_FilterDesignTechniques/Middle_WindowEffect/test_original_filter_in_c.py
import numpy as np
from bandpassfilter import bandpassfilter
from filter_analysis_imp import filter_analysis_imp

fps = 30
LPF = 0.5
HPF = 3.33

n = np.arange(-511, 513)
imp = (n == 0).astype(float)

imp_rep = bandpassfilter(imp, 128, LPF, HPF, fps)
imp_rep = np.concatenate([np.zeros(2048), imp_rep, np.zeros(2048)])

filter_analysis_imp(imp_rep, fps, "Ideal BPF with windowing")
